infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesWorkspaceServiceAccount.java

```diff
@@ -50,14 +53,19 @@ public KubernetesWorkspaceServiceAccount(
 
   @Override
   protected Role buildRole(
-      String name, List<String> resources, List<String> apiGroups, List<String> verbs) {
+      String name,
+      List<String> resources,
+      @Nullable List<String> resourceNames,
+      List<String> apiGroups,
+      List<String> verbs) {
     return new RoleBuilder()
         .withNewMetadata()
         .withName(name)
         .endMetadata()
         .withRules(
             new PolicyRuleBuilder()
                 .withResources(resources)
+                .withResourceNames(resourceNames != null ? resourceNames : emptyList())
```

Comment:
is the functionality equivalent between setting the emptyList() and don't set `ResourceNames` at all ?

Reply:
You are right, but in that case the method won't be in a single row format :)

Reply:
I'm really asking if it is effectively equivalent. Empty `resourceName` list might evoke that this role is is not applied to any resource name. But I guess goal here is if list is empty, to apply it to all resource names.

Reply:
No, empty list won't set any limitations at all, so empty list is equivalent to don't set the `ResourceNames` at all.